examples/tv-casting-app/android/App/app/src/main/java/com/chip/casting/app/CommissionerDiscoveryFragment.java

```diff
@@ -8,143 +8,183 @@
-   */
-  public static CommissionerDiscoveryFragment newInstance(TvCastingApp tvCastingApp) {
-    return new CommissionerDiscoveryFragment(tvCastingApp);
-  }
-
-  @Override
-  public void onCreate(Bundle savedInstanceState) {
-    super.onCreate(savedInstanceState);
-  }
-
-  @Override
-  public View onCreateView(
-      LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
-    // Inflate the layout for this fragment
-    return inflater.inflate(R.layout.fragment_commissioner_discovery, container, false);
-  }
-
-  @Override
-  public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
-    super.onViewCreated(view, savedInstanceState);
-
-    Button manualCommissioningButton = getView().findViewById(R.id.manualCommissioningButton);
-    Callback callback = (Callback) this.getActivity();
-    View.OnClickListener manualCommissioningButtonOnClickListener =
-        new View.OnClickListener() {
-          @Override
-          public void onClick(View v) {
-            callback.handleCommissioningButtonClicked(null);
-          }
-        };
-    manualCommissioningButton.setOnClickListener(manualCommissioningButtonOnClickListener);
-
-    Context context = this.getContext();
-    SuccessCallback<DiscoveredNodeData> successCallback =
-        new SuccessCallback<DiscoveredNodeData>() {
-          @Override
-          public void handle(DiscoveredNodeData discoveredNodeData) {
-            Log.d(TAG, "Discovered a Video Player Commissioner: " + discoveredNodeData);
-            String buttonText = getCommissionerButtonText(discoveredNodeData);
-
-            if (!buttonText.isEmpty()) {
-              Button commissionerButton = new Button(context);
-              commissionerButton.setText(buttonText);
-              CommissionerDiscoveryFragment.Callback callback =
-                  (CommissionerDiscoveryFragment.Callback) getActivity();
-              commissionerButton.setOnClickListener(
-                  new View.OnClickListener() {
-                    @Override
-                    public void onClick(View v) {
-                      Log.d(
-                          TAG,
-                          "CommissionerResolveListener.onServiceResolved.OnClickListener.onClick called for "
-                              + discoveredNodeData);
-                      callback.handleCommissioningButtonClicked(discoveredNodeData);
-                    }
-                  });
-              new Handler(Looper.getMainLooper())
-                  .post(
-                      () ->
-                          ((LinearLayout) getActivity().findViewById(R.id.castingCommissioners))
-                              .addView(commissionerButton));
+    private static final String TAG = CommissionerDiscoveryFragment.class.getSimpleName();
+    private static final long DISCOVERY_POLL_INTERVAL_MS = 10000;
+    private static final ArrayList<DiscoveredNodeData> commissionablePlayerList = new ArrayList<>();
```

Comment:
Note that the video player is a commissioner, not a commissionable node. Can you change this name to commissionerVideoPlayerList? Also, List< DiscoveredNodeData> is preferred over ArrayList< DiscoveredNodeData> for the reference type